### PR TITLE
Flutter v2

### PR DESCRIPTION
**Pull request by khatruong2009**
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Comment by hdworld11** [comment]
Storage section lgtm

File: src/fragments/lib/project-setup/flutter/upgrade-guide/upgrade-guide.mdx [hunk 1]
```diff
@@ -1,328 +1,173 @@
-With the release of v1 Amplify Flutter now supports Web and Desktop for Auth, API, Analytics, and Storage use cases. Developers can build cross-platform Flutter apps with Amplify that target iOS, Android, Web, and Desktop (macOS, Windows, Linux) using a single codebase.
+Amplify Flutter v2 has several changes that may require migration when upgrading from v1.
```
Comment (salihgueler) [inline src/fragments/lib/project-setup/flutter/upgrade-guide/upgrade-guide.mdx:1]: Can we mention here and manage expectation on what kind of changes user should expect? E.g. "Amplify Flutter v2 has changes over Auth, Data and Storage to do xxx"

File: src/fragments/lib/project-setup/flutter/upgrade-guide/upgrade-guide.mdx [hunk 2]
```diff
@@ -1,328 +1,173 @@
...
 
-We have re-written our libraries in Dart. In some places, we have made breaking changes to improve ergonomics or enable features that had been missing from the v0 implementations.
+## Auth
```
Comment (salihgueler) [inline src/fragments/lib/project-setup/flutter/upgrade-guide/upgrade-guide.mdx:3]: Auth or Authentication?

File: src/fragments/lib/project-setup/flutter/upgrade-guide/upgrade-guide.mdx [hunk 3]
```diff
@@ -1,328 +1,173 @@
...
 
-## Prerequisites
+#### Time based OTP Exception Handling
```
Comment (salihgueler) [inline src/fragments/lib/project-setup/flutter/upgrade-guide/upgrade-guide.mdx:5]: I would recommend either using TOTP or Time Based One Time Password (or better use both)

File: src/fragments/lib/project-setup/flutter/upgrade-guide/upgrade-guide.mdx [hunk 4]
```diff
@@ -1,328 +1,173 @@
...
 
-- A Flutter application targeting Flutter SDK >= 3.3.0 with Amplify libraries integrated
+In Amplify Flutter v1, `Amplify.Auth.verifyTotpSetup()` throws an `EnableSoftwareTokenMfaException` if the provided code was incorrect. In Amplify Flutter v2 a `CodeMismatchException` is thrown.
```
Comment (salihgueler) [inline src/fragments/lib/project-setup/flutter/upgrade-guide/upgrade-guide.mdx:7]: nit: First sentence uses active voice and second sentence uses passive voice. Let's use active in both.

File: src/fragments/lib/project-setup/flutter/upgrade-guide/upgrade-guide.mdx [hunk 5]
```diff
@@ -1,328 +1,173 @@
...
 
-Amplify requires a minimum target platform for iOS (13.0), Android (API level 24), and macOS (10.15). Additional setup is required for some target platforms. Please see the [platform setup](/[platform]/start/project-setup/platform-setup/) guide for more details on platform-specific setup.
+#### Auth Flow in Amplify Config
 
-## Platform Setup
+In Amplify Flutter v1, the value of `“authenticationFlowType”` from the amplify configuration object is used as the default `AuthenticationFlowType` if none is provided when calling `Amplify.Auth.signIn()`. In Amplify Flutter v2 `AuthenticationFlowType.userSrpAuth` is the default if none is provided. To change the authentication flow type, provide a value in the sign in options.
```
Comment (salihgueler) [inline src/fragments/lib/project-setup/flutter/upgrade-guide/upgrade-guide.mdx:11]: nit: amplify --> Amplify


```suggestion
In Amplify Flutter v1, the value of `“authenticationFlowType”` from the Amplify configuration object is used as the default `AuthenticationFlowType` if none is provided when calling `Amplify.Auth.signIn()`. 

In Amplify Flutter v2, default type is `AuthenticationFlowType.userSrpAuth`. To change the authentication flow type, you can provide a value in the sign in options under `options` parameter..
```

File: src/fragments/lib/project-setup/flutter/upgrade-guide/upgrade-guide.mdx [hunk 6]
```diff
@@ -1,328 +1,173 @@
...
-  // the access and/or id token is expired but cannot be refreshed because the
-  // user is offline.
-}
-
-// `AuthResult.valueOrNull` will return the identityId if present or null,
-// but will never throw even if the underlying operation failed.
-final identityId = session.identityIdResult.valueOrNull;
-safePrint('identityId: $identityId');
-```
+In Amplify Flutter v1, the autoFlushEventsInterval value was configured manually in the amplifyconfiguration.dart file. In Amplify Flutter v2, the autoFlushEventsInterval value is not read from amplifyconfiguration.dart but instead is passed directly to the `AmplifyAnalyticsPinpoint()` constructor under the `AnalyticsPinpointPluginOptions`. The default value for autoFlushEventsInterval remains at 30 seconds, consistent with the behavior in Amplify Flutter v1.
```
Comment (salihgueler) [inline src/fragments/lib/project-setup/flutter/upgrade-guide/upgrade-guide.mdx:30]: Same active/passive voice inconsistency is here too. 

```suggestion
In Amplify Flutter v1, the `autoFlushEventsInterval` value was configured manually in the `amplifyconfiguration.dart` file. In Amplify Flutter v2, the `autoFlushEventsInterval` value is not read from `amplifyconfiguration.dart` but instead is passed directly to the `AmplifyAnalyticsPinpoint()` constructor under the `AnalyticsPinpointPluginOptions`. The default value for `autoFlushEventsInterval` remains at 30 seconds, consistent with the behavior in Amplify Flutter v1.
```

File: src/fragments/lib/project-setup/flutter/upgrade-guide/upgrade-guide.mdx [hunk 7]
```diff
@@ -1,328 +1,173 @@
...
-switch (result.nextStep.signUpStep) {
-  case AuthSignUpStep.confirmSignUp:
-    // prompt user to confirm sign in using nextStep.codeDeliveryDetails
-    break;
-  case AuthSignUpStep.done:
-    // sign up is complete, user can be taken to an authenticated state.
-    break;
-}
-```
+In Amplify Flutter v1, files operations were performed based on a key and an AccessLevel. Amplify translated these values to a full file path. The AccessLevel (guest, protected, or private) determined what prefix to use (`“public/”`, `“protected/<user_identity_id>/”`, or `“private/<user_identity_id>/”`). Amplify Flutter v1 forms the full path using the prefix and the provided key.
```
Comment (salihgueler) [inline src/fragments/lib/project-setup/flutter/upgrade-guide/upgrade-guide.mdx:36]:
```suggestion
In Amplify Flutter v1, files operations were performed based on a `key` and an `AccessLevel`. Amplify translated these values to a full file path. The `AccessLevel` (guest, protected, or private) determined what prefix to use (`“public/”`, `“protected/<user_identity_id>/”`, or `“private/<user_identity_id>/”`). Amplify Flutter v1 forms the full path using the prefix and the provided key.
```

File: src/fragments/lib/project-setup/flutter/upgrade-guide/upgrade-guide.mdx [hunk 8]
```diff
@@ -1,328 +1,173 @@
...
+
+Amplify Flutter v2 allows for full control of the path of the storage object. Storage paths can be constructed either from a static string or from the current users identity id using the StoragePath class.
+
+To migrate from v1 to v2, replace all uses of `key` with `path` and remove uses of `StorageAccessLevel`. The path should include the prefix that was previously added by Amplify automatically.
 
-- Auth hub events now have a specific subtype (`AuthHubEvent`) and contain a `type` that is an enum. See the example below for how to migrate.
+##### Example for Public Files
```
Comment (salihgueler) [inline src/fragments/lib/project-setup/flutter/upgrade-guide/upgrade-guide.mdx:42]:
```suggestion
##### Example Migration for Public Files
```

File: src/fragments/lib/project-setup/flutter/upgrade-guide/upgrade-guide.mdx [hunk 9]
```diff
@@ -1,328 +1,173 @@
...
-});
+// upload file to "public/file.txt" in Amplify Flutter v2
+Amplify.Storage.uploadFile(
+  localFile: file,
+  path: const StoragePath.fromString("public/file.txt"),
+);
 ```
 
-- Cognito-specific API options such as `CognitoSignInOptions` are deprecated in favor of `pluginOptions`. The classes will be removed in a future release. See below for an example of how to migrate for `Auth.signIn()`.
+##### Example for Protected Files
```
Comment (salihgueler) [inline src/fragments/lib/project-setup/flutter/upgrade-guide/upgrade-guide.mdx:63]:
```suggestion
##### Example Migration for Protected Files
```

File: src/fragments/lib/project-setup/flutter/upgrade-guide/upgrade-guide.mdx [hunk 10]
```diff
@@ -1,328 +1,173 @@
...
+  path: StoragePath.fromIdentityId(
+    (identityId) => "private/$identityId/file.txt",
+  ),  
+);
+```
 
-<Block name="iOS">
+#### Prefix Resolver has been removed
 
+With the introduction of StoragePath, there is no need to override the libraries behavior for the prefix.
```
Comment (salihgueler) [inline src/fragments/lib/project-setup/flutter/upgrade-guide/upgrade-guide.mdx:111]:
```suggestion
With the introduction of `StoragePath`, there is no need to override the libraries behavior for the prefix. 
```

File: src/fragments/lib/project-setup/flutter/upgrade-guide/upgrade-guide.mdx [hunk 11]
```diff
@@ -1,328 +1,173 @@
...
-          <category android:name="android.intent.category.BROWSABLE" />
-          <data android:scheme="myapp" />
-      </intent-filter>
-  </activity>
-  ...
-</application>
-```
-
-Start by removing this code, then update the `AndroidManifest.xml` as follows, replacing `myapp` with your custom URI scheme you configured in your backend.
+##### Example for Private Files
```
Comment (Jordan-Nelson) [inline src/fragments/lib/project-setup/flutter/upgrade-guide/upgrade-guide.mdx:86]:
```suggestion
##### Example Migration for Private Files
```

**Pull request closed without merging**